lib/Doctrine/ODM/MongoDB/Aggregation/Operator/GenericOperatorsInterface.php

```diff
@@ -0,0 +1,1356 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ODM\MongoDB\Aggregation\Operator;
+
+use Doctrine\ODM\MongoDB\Aggregation\Expr;
+
+/**
+ * Interface containing all operators available in most pipeline stages.
+ * This interface can be used for type hinting, but must not be implemented by
+ * users. Methods WILL be added to the public API in future minor versions.
+ *
+ * @internal
+ */
+interface GenericOperatorsInterface
+{
+    /**
+     * Returns the absolute value of a number.
+     *
+     * The <number> argument can be any valid expression as long as it resolves
+     * to a number.
+     *
+     * @see https://docs.mongodb.com/manual/reference/operator/aggregation/abs/
+     *
+     * @param mixed|Expr $number
+     */
+    public function abs($number): self;
```

Comment:
Here and all methods below - shouldn't we make it something like:
```suggestion
     * @param static $number
     */
    public function abs($number): static;
```

Since we are targeting PHP 7.4+, we can use `static` typehint. We shouldn't have any reference to `Expr` in this interface.

Reply:
I was considering that, but this was added in 8.0, not 7.4: https://php.watch/versions/8.0/static-return-type.

Since the interface is internal, we can make this change once we drop support for 7.4. 👍

Reply:
Makes sense, thanks!